### PR TITLE
feat: Service Start Page

### PR DESCRIPTION
**Pull request by rhysbarrett**
### JIRA link

https://companieshouse.atlassian.net/browse/SR-23

### Change description

This commit implements and tests the contents of the Service Start Page. It does not include navigation.

### Page Screenshot (if applicable)

chs local_suppress-my-details

### Work Checklist

- [x] Tests added where applicable
- [x] UI changes look good on mobile
- [x] UI changes meet accessibility criteria

---

### Merge Instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+
+
+### Work Checklist
+
+- [ ] Tests added where applicable
+- [ ] UI changes look good on mobile
+- [ ] UI changes meet accessibility criteria
+
+---
```
Comment (akaJSaunders) [inline .github/PULL_REQUEST_TEMPLATE.md:19]: Thanks for adding this template, where did you get it from? The reason I ask is because I never really found this work checklist to be very usefull

Comment (rhysbarrett) [reply .github/PULL_REQUEST_TEMPLATE.md:19]: Nabbed it from Appeals :) I've modified some parts slightly (such as adding the reminder to add a screenshot) but left the checklist be, as I do find it pretty useful for reminding me to do those basic mobile/accessibility checks.

File: src/controllers/StartPageController.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { NextFunction, Request, Response } from 'express';
+
+export class StartPageController {
+
+  public renderView = (req: Request, res: Response, next: NextFunction) => res.render('index');
```
Comment (akaJSaunders) [inline src/controllers/StartPageController.ts:5]: I see the name of the template is still 'index', I think we can change that to something more relevant. What about 'landing-page'

Comment (rhysbarrett) [reply src/controllers/StartPageController.ts:5]: Good idea. I prefer 'start-page' though as it's consistent with the Controller naming and GDS terminology.

File: src/views/index.njk [hunk 3]
```diff
@@ -1,13 +1,76 @@
...
+
+{% block pageTitle %}
+  {{ serviceName }}
 {% endblock %}
 
 {% block content %}
-    <h1 class="govuk-heading-xl">Hello World</h1>
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+      <h1 class="govuk-heading-xl">{{ serviceName }}</h1>
```
Comment (akaJSaunders) [inline src/views/index.njk:13]:
```suggestion
      <h1 class="govuk-heading-xl"> {{ serviceName }} </h1>
```

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (akaJSaunders) [reply .github/PULL_REQUEST_TEMPLATE.md:19]: Okay fair enough, maybe I need to pay attention to them a bit more haha

File: src/views/layout.njk [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+{% from 'govuk/components/header/macro.njk'       import govukHeader %}
+{% from 'govuk/components/phase-banner/macro.njk' import govukPhaseBanner %}
+
+{% set serviceName = 'Apply to remove your home address from the Companies House register' %}
+
+{% block head %}
+  <!--[if !IE 8]><!-->
+  <link href="//{{ CDN_HOST }}/stylesheets/govuk-frontend/v3.6.0/govuk-frontend-3.6.0.min.css" rel="stylesheet"/>
+  <link href="//{{ CDN_HOST }}/stylesheets/services/suppression/application.css" rel="stylesheet"/>
+  <!--<![endif]-->
```
Comment (akaJSaunders) [inline src/views/layout.njk:13]: Could you indent the links, as they are part of that if statement

File: src/views/layout.njk [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  }}
+{% endblock %}
+
+{% set mainClasses = mainClasses | default('govuk-main-wrapper--auto-spacing') %}
+
+{% block beforeContent %}
+  {{
+    govukPhaseBanner({
+      tag: {
+        text: 'Beta'
```
Comment (akaJSaunders) [inline src/views/layout.njk:33]: Should be alpha according to screenshot in the ticket. I know we had a discussion around that, but no notes state that it should be beta, so lets just be safe.

Comment (rhysbarrett) [reply src/views/layout.njk:33]: It was noted in Refinement that the screenshot is incorrect. I'll double check with Bart.

File: src/views/layout.njk [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+            text: 'Contact us'
+          },
+          {
+            href: 'https://developer.companieshouse.gov.uk/',
+            text: 'Developers'
+          }
+        ]
+      }
+    })
+  }}
```
Comment (akaJSaunders) [inline src/views/layout.njk:65]: This is out of scope. Don't think we need to add anything here, lets avoid making assumptions. Could we perhaps have an empty footer?

File: src/views/layout.njk [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+{% block beforeContent %}
+  {{
+    govukPhaseBanner({
+      tag: {
+        text: 'Beta'
+      },
+      html: 'This is a new service – your <a class="govuk-link" href="#" target="_blank">feedback</a> will help us to improve it.'
+    })
+  }}
+  {% block backLink %}{% endblock %}
```
Comment (akaJSaunders) [inline src/views/layout.njk:38]: Dont think we need this
```suggestion
```

File: src/views/layout.njk (hunk 5 above)
Comment (rhysbarrett) [reply src/views/layout.njk:33]: Confirmed by Bart, Beta is the correct phase.

File: test/controllers/StartPageController.test.ts [hunk 8]
```diff
@@ -0,0 +1,30 @@
+
```
Comment (akaJSaunders) [inline test/controllers/StartPageController.test.ts:1]:
```suggestion
```
Yes, I am merciless!

File: src/views/layout.njk (hunk 7 above)
Comment (rhysbarrett) [reply src/views/layout.njk:38]: Good spot.  Removed.

File: src/views/index.njk [hunk 9]
```diff
@@ -1,13 +1,76 @@
...
+
+      {{
+        govukButton({
+          text: 'Start now',
+          href: '#',
+          classes: 'govuk-!-margin-top-2 govuk-!-margin-bottom-8',
+          isStartButton: true
+        })
+      }}
+    </div>
```
Comment (akaJSaunders) [inline src/views/index.njk:56]:
```suggestion
    </div>
    
```

File: test/controllers/StartPageController.test.ts (hunk 8 above)
Comment (rhysbarrett) [reply test/controllers/StartPageController.test.ts:1]: No that's a very good spot, it's not at all obvious in my editor.

File: src/views/index.njk (hunk 9 above)
Comment (akaJSaunders) [inline src/views/index.njk:56]:
```suggestion
    </div>
    
```

File: src/views/index.njk [hunk 10]
```diff
@@ -1,13 +1,76 @@
...
+          <ul class="govuk-list govuk-body-s">
+            <li>
+              <a class="govuk-link" target="_blank" href="https://www.gov.uk/government/publications/restricting-the-disclosure-of-your-psc-information">
+                Restricting the disclosure of your information
+              </a>
+            </li>
+          </ul>
+        </nav>
+      </aside>
+    </div>
```
Comment (akaJSaunders) [inline src/views/index.njk:72]:
```suggestion
    </div>
    
```

File: src/views/layout.njk (hunk 5 above)
Comment (akaJSaunders) [reply src/views/layout.njk:33]: thank you

File: test/controllers/StartPageController.test.ts [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+        it('should return 200 and render the Service Start Page', async () => {
+            const expectedPatterns = [
+                /Apply to remove your home address from the Companies House register/,
+                /Use this service to apply to remove your home address, or part of it, from the Companies House public register \(SR01\)\./,
+                /<a.*>\s*Restricting the disclosure of your information\s*<\/a>/,
+                /<a.* href="https:\/\/www.gov.uk\/government\/publications\/restricting-the-disclosure-of-your-psc-information">.*<\/a>/s,
+                /<a.*>\s*Companies House register\s*<\/a>/,
+                /<a.* href="https:\/\/beta.companieshouse.gov.uk\/">.*<\/a>/s,
+            ]
```
Comment (akaJSaunders) [inline test/controllers/StartPageController.test.ts:19]:
```suggestion
            ];
```
